app/controllers/application_controller.rb

```diff
@@ -10,22 +10,23 @@ class ApplicationController < ActionController::Base
   unless Rails.application.config.consider_all_requests_local
     rescue_from ActiveRecord::RecordNotFound, :with => :record_not_found
     rescue_from ActionController::RoutingError, :with => :route_not_found
+    rescue_from ActionController::UnknownController, :with => :route_not_found
```

Comment:
hah, I see what was happening:
```
redirect_to controller: "/home", action: "index"
```

`controller: "/home"` is invalid, because there is no `/home_controller.rb` its `home_controller`. Thus ActionController throws an `UnknownController` exception. You're catching this and then rethrowing a `route_not_found`. Which is then caught and routed back to `root_path`.